perfkitbenchmarker/configs/__init__.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Configuration files for benchmarks."""
+
+import copy
+import yaml
+
+from perfkitbenchmarker import data
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+
+FLAGS = flags.FLAGS
+CONFIG_CONSTANTS = 'default_config_constants.yaml'
+
+flags.DEFINE_string('benchmark_config_file', None,
+                    'The file path to the user config file which will '
+                    'override benchmark defaults.')
+flags.DEFINE_multistring(
+    'config_override', None,
+    'This flag can be used to override any config value. It is applied after '
+    'the user config (specified via --benchmark_config_file_path), so it has '
+    'a higher priority than that config. The value of the flag should be '
+    'fully.qualified.key=value (e.g. --config_override=cluster_boot.vm_groups.'
+    'default.vm_count=4). This flag can be repeated.')
+
+
+def _LoadUserConfig(path):
+  """Loads a user config from the supplied path."""
+  with open(data.ResourcePath(path)) as fp:
```

Comment:
Works for absolute paths and paths relative to the cwd but not paths using '~'. I updated the flag documentation